smart_control/simulator/enhanced_occupancy_test.py

```diff
@@ -0,0 +1,259 @@
+    expected_occasional = 1000 * TEST_SETUP_OCCASIONAL_PCT
+    expected_weekday = 1000 * (
+        1.0 - TEST_SETUP_REGULAR_PCT - TEST_SETUP_OCCASIONAL_PCT
+    )
+    self.assertAlmostEqual(
+        distribution['weekday_only'], expected_weekday, delta=100
+    )
+    self.assertAlmostEqual(
+        distribution['weekend_regular'], expected_regular, delta=100
+    )
+    self.assertAlmostEqual(
+        distribution['weekend_occasional'], expected_occasional, delta=100
+    )
+
+  def test_parameter_variation(self):
+    occupant = EnhancedZoneOccupant(
+        earliest_expected_arrival_min=EARLIEST_EXPECTED_ARRIVAL_HOUR * 60,
+        latest_expected_arrival_min=LATEST_EXPECTED_ARRIVAL_HOUR * 60,
+        earliest_expected_departure_min=EARLIEST_EXPECTED_DEPARTURE_HOUR * 60,
+        latest_expected_departure_min=LATEST_EXPECTED_DEPARTURE_HOUR * 60,
+        lunch_start_min=LUNCH_START_HOUR * 60,
+        lunch_end_min=LUNCH_END_HOUR * 60,
+        step_size=TIME_STEP,
+        random_state=np.random.RandomState(seed=SEED),
+        time_zone=DEFAULT_TIMEZONE,
+        worker_type=WorkerType.WEEKDAY_ONLY,
+        weekend_work_prob=NO_WEEKEND_WORKERS_DAILY_PROB,
+        occupant_id=0,
+    )
+    day1_morning = pd.Timestamp('2021-09-01 09:00')
+    day1_afternoon = pd.Timestamp('2021-09-01 15:00')
+    day2_morning = pd.Timestamp('2021-09-02 09:00')
+    params1_morning = occupant._get_daily_params(day1_morning)
+    params1_afternoon = occupant._get_daily_params(day1_afternoon)
+    params2_morning = occupant._get_daily_params(day2_morning)
+    self.assertEqual(params1_morning, params1_afternoon)
+    self.assertNotEqual(params1_morning, params2_morning)
+
+  @parameterized.parameters(None, 'UTC', 'US/Eastern', 'US/Pacific')
+  def test_occupant_peek(self, tz):
+    occupant = EnhancedZoneOccupant(
+        earliest_expected_arrival_min=EARLIEST_EXPECTED_ARRIVAL_HOUR * 60,
+        latest_expected_arrival_min=LATEST_EXPECTED_ARRIVAL_HOUR * 60,
+        earliest_expected_departure_min=EARLIEST_EXPECTED_DEPARTURE_HOUR * 60,
+        latest_expected_departure_min=LATEST_EXPECTED_DEPARTURE_HOUR * 60,
+        lunch_start_min=LUNCH_START_HOUR * 60,
+        lunch_end_min=LUNCH_END_HOUR * 60,
+        step_size=TIME_STEP,
+        random_state=np.random.RandomState(seed=SEED),
+        time_zone=tz,
+        worker_type=WorkerType.WEEKDAY_ONLY,
+        weekend_work_prob=NO_WEEKEND_WORKERS_DAILY_PROB,
+        occupant_id=0,
+    )
+    day1_early_morning = pd.Timestamp('2021-09-01 06:00')
+    day1_work_morning = pd.Timestamp('2021-09-01 10:00')
+    day1_afternoon = pd.Timestamp('2021-09-01 15:00')
+    day1_evening = pd.Timestamp('2021-09-01 20:00')
+    weekend = pd.Timestamp('2021-09-05 08:00')
+
+    self.assertEqual(occupant.peek(day1_early_morning), OccupancyStateEnum.AWAY)
+    self.assertEqual(occupant.peek(day1_work_morning), OccupancyStateEnum.WORK)
+    self.assertEqual(occupant.peek(day1_afternoon), OccupancyStateEnum.WORK)
```

Comment:
looks like there might be a test failure around here. please check.

https://github.com/google/sbsim/actions/runs/17325401231/job/49452306954?pr=123#step:11:208